apps/gitness/src/pages-v2/repo/repo-summary.tsx

```diff
@@ -0,0 +1,225 @@
+import { TokenSuccessDialog } from '../../pages/profile-settings/token-create/token-success-dialog'
+import { PathParams } from '../../RouteDefinitions'
+import { decodeGitContent, getTrimmedSha, normalizeGitRef } from '../../utils/git-utils'
+
+export default function RepoSummaryPage() {
+  const [loading, setLoading] = useState(false)
+  const [files, setFiles] = useState<FileProps[]>([])
+  const repoRef = useGetRepoRef()
+  const navigate = useNavigate()
+  const { spaceId, repoId, gitRef } = useParams<PathParams>()
+
+  const { data: { body: repository } = {} } = useFindRepositoryQuery({ repo_ref: repoRef })
+
+  const { data: { body: branches } = {} } = useListBranchesQuery({
+    repo_ref: repoRef,
+    queryParams: { include_commit: false, sort: 'date', order: 'asc', limit: 20, page: 1, query: '' }
+  })
+
+  const [selectedBranch, setSelectedBranch] = useState<string>('')
+  const [createdTokenData, setCreatedTokenData] = useState<(TokenFormType & { token: string }) | null>(null)
+  const [successTokenDialog, setSuccessTokenDialog] = useState(false)
+
+  const { query } = useCommonFilter()
+
+  const branchList = useMemo(() => {
+    return (
+      branches?.map(item => ({
+        name: item?.name || ''
+      })) || []
+    )
+  }, [branches])
+
+  const { data: { body: repoSummary } = {} } = useSummaryQuery({
+    repo_ref: repoRef,
+    queryParams: { include_commit: false, sort: 'date', order: 'asc', limit: 20, page: 1, query }
+  })
+
+  const { branch_count, default_branch_commit_count, pull_req_summary, tag_count } = repoSummary || {}
+
+  const { data: { body: readmeContent } = {} } = useGetContentQuery({
+    path: 'README.md',
+    repo_ref: repoRef,
+    queryParams: { include_commit: false, git_ref: normalizeGitRef(selectedBranch) }
+  })
+
+  const decodedReadmeContent = useMemo(() => {
+    return decodeGitContent(readmeContent?.content?.data)
+  }, [readmeContent])
+
+  const { data: { body: repoDetails } = {} } = useGetContentQuery({
+    path: '',
+    repo_ref: repoRef,
+    queryParams: { include_commit: true, git_ref: normalizeGitRef(selectedBranch) }
+  })
+
+  const { mutate: createToken } = useCreateTokenMutation(
+    { body: {} },
+    {
+      onSuccess: ({ body: newToken }) => {
+        const tokenData = {
+          identifier: newToken.token?.identifier ?? 'Unknown',
+          lifetime: newToken.token?.expires_at
+            ? new Date(newToken.token.expires_at).toLocaleDateString()
+            : 'No Expiration',
+          token: newToken.access_token ?? 'Token not available'
+        }
+
+        setCreatedTokenData(tokenData)
+        setSuccessTokenDialog(true)
+      }
+    }
+  )
+
+  const handleCreateToken = () => {
+    const body = {
+      identifier: `code_token_${generateAlphaNumericHash(5)}`
+    }
+    createToken({ body })
+  }
+
+  const repoEntryPathToFileTypeMap = useMemo((): Map<string, OpenapiGetContentOutput['type']> => {
```

Comment:
Not an issue, but typing the memo or the variable is more consistent then typing the function return in this case.
```suggestion
  const repoEntryPathToFileTypeMap = useMemo<Map<string, OpenapiGetContentOutput['type']>>(() => {
```

or

```suggestion
  const repoEntryPathToFileTypeMap: Map<string, OpenapiGetContentOutput['type']> = useMemo(() => {
```